### PR TITLE
Testing branch

### PR DESCRIPTION
**Pull request by Chris-Middleton**
I added testing methods for the backend and rewrote some backend code to get them to pass. Also did a small change to frontend/createPost so that all fields are included in the json object sent to the server.

### CODE REVIEW

**Comment by Chris-Middleton** [comment]
Okay travis tests failed. Dont merge this yet, I'm gonna work on fixing it.

**Comment by dreko1** [comment]
I like these tests and once Travis is passing I think that this will be perfect to merge. Good work!

**Comment by Chris-Middleton** [comment]
Woohoo it passed! I think we need to increase backend code coverage, and some tests aren't yet written, but this pull request can be merged now.

**Comment by Cheeems** [comment]
I think this looks good to merge to me! Great work

**Comment by ndiekema** [comment]
Nice work on the tests, it looks good to commit! I can help write more tests for full code coverage tomorrow.

File: backend/backend.py [hunk 1]
```diff
@@ -3,13 +3,13 @@
...
 from flask import Flask, request, flash, jsonify, redirect, url_for
-from flask_login import (
+'''from flask_login import (
     LoginManager,
     current_user,
     login_required,
     login_user,
     logout_user
-)
+)'''
```
Comment (ndiekema) [inline backend/backend.py:12]: I'm currently working on implementing user-authentication with Flask-Login so we will need this.

File: backend/backend.py [hunk 2]
```diff
@@ -29,114 +29,93 @@
...
-            }
-            user_to_add = User(user_to_add)
-            user_to_add.save()
-            return jsonify(new_user), 201
-
-
-# this method will need to be called by the method the queries the database
-def validate_password(pw, db_pw):
-    return bcrypt.checkpw(pw.encode('utf8'), db_pw)
+        if not (req['username'] and req['password'] and req['email']):
```
Comment (ndiekema) [inline backend/backend.py:47]: I changed some of this in my commit as well but I like your approach better. We'll just have to deal with a couple merge conflicts

File: backend/model.py [hunk 3]
```diff
@@ -2,270 +2,195 @@
...
-    #         resp = self.collection.update_one(
-    #             {"_id": ObjectId(self._id)}, self)
-    #     self._id = str(self._id)
-    #     return resp
-
-    def get(self, username):
-        user = self.collection.find_one({"username": username})
+    '''
+
+    def new_user(username, password, email):
```
Comment (ndiekema) [inline backend/model.py:75]: Good idea moving this to the db model.

**Review by ndiekema — APPROVED** [review]
This is much better organized, nice work.

**Pull request merged**